mmv1/third_party/terraform/services/pubsub/resource_pubsub_subscription_test.go

```diff
@@ -681,40 +685,40 @@ resource "google_service_account" "bq_write_service_account" {
 }
 
 resource "google_project_iam_member" "viewer" {
-	project = data.google_project.project.project_id
-	role   = "roles/bigquery.metadataViewer"
-	member = "serviceAccount:${google_service_account.bq_write_service_account.email}"
+  project = data.google_project.project.project_id
+  role    = "roles/bigquery.metadataViewer"
+  member  = "serviceAccount:${google_service_account.bq_write_service_account.email}"
 }
 
 resource "google_project_iam_member" "editor" {
-	project = data.google_project.project.project_id
-	role   = "roles/bigquery.dataEditor"
-	member = "serviceAccount:${google_service_account.bq_write_service_account.email}"
+  project = data.google_project.project.project_id
+  role    = "roles/bigquery.dataEditor"
+  member  = "serviceAccount:${google_service_account.bq_write_service_account.email}"
 }`, serviceAccountId)
 		serviceAccountEmailField = "service_account_email = google_service_account.bq_write_service_account.email"
 	} else {
 		serviceAccountResource = fmt.Sprintf(`
 resource "google_project_iam_member" "viewer" {
```

Comment:
@melinath or @NickElliot , do you know if this resource should be removed with the bootstrapped IAM?

Reply:
I think in this case, it can't be, because there are a couple test cases where the service account is created _within_ the test case, so in those cases, the permission still needs to be added. 

For example
https://github.com/GoogleCloudPlatform/magic-modules/blob/01a6cac159ae042b73ec49c4ef80eb29ffd5098f/mmv1/third_party/terraform/services/pubsub/resource_pubsub_subscription_test.go#L251
where it creates `bq-test-sa` (which will later be destroyed), and then `bq-test-sa2` on the 3rd test case.

Since the account will be created and destroyed within the test, it wouldn't be possible to boostrap its permissions.

However, if the test is flaky (or still flaky), it would be possible to add a sleep in there.

From what I can see, in the case where the permissions already exist from bootstrapping, Terraform doesn't seem to complain (maybe the test framework imports them when they exist?)

Reply:
I see.  `testAccPubsubSubscriptionBigQuery_basic` handles two cases, no-empty service account and empty service account.

I think it is OK to not add a sleep right now. If it is still flaky, we can add the sleep later.

Reply:
Right. We could easily enough not define the resources when the service account isn't created... but there's an explicit `depends` later that I think would be harder to get rid of....

I could reproduce flakiness / permissions issue with some of the test cases locally. This is what the sleep would look like FWIW.

```diff
diff --git a/mmv1/third_party/terraform/services/pubsub/resource_pubsub_subscription_test.go b/mmv1/third_party/terraform/services/pubsub/resource_pubsub_subscription_test.go
index 6e2447ff4..5f2fe23db 100644
--- a/mmv1/third_party/terraform/services/pubsub/resource_pubsub_subscription_test.go
+++ b/mmv1/third_party/terraform/services/pubsub/resource_pubsub_subscription_test.go
@@ -246,6 +246,9 @@ func TestAccPubsubSubscriptionBigQuery_serviceAccount(t *testing.T) {
 		PreCheck:                 func() { acctest.AccTestPreCheck(t) },
 		ProtoV5ProviderFactories: acctest.ProtoV5ProviderFactories(t),
 		CheckDestroy:             testAccCheckPubsubSubscriptionDestroyProducer(t),
+		ExternalProviders: map[string]resource.ExternalProvider{
+			"time": {},
+		},
 		Steps: []resource.TestStep{
 			{
 				Config: testAccPubsubSubscriptionBigQuery_basic(dataset, table, topic, subscriptionShort, false, "bq-test-sa"),
@@ -715,6 +718,11 @@ resource "google_project_iam_member" "editor" {
 	return fmt.Sprintf(`
 data "google_project" "project" {}
 
+resource "time_sleep" "wait_30_seconds" {
+  create_duration = "30s"
+}
+
+
 %s
 
 resource "google_bigquery_dataset" "test" {
@@ -754,7 +762,8 @@ resource "google_pubsub_subscription" "foo" {
 
   depends_on = [
     google_project_iam_member.viewer,
-    google_project_iam_member.editor
+    google_project_iam_member.editor,
+    time_sleep.wait_30_seconds,
   ]
 }
 	`, serviceAccountResource, dataset, table, topic, subscription, useTableSchema, serviceAccountEmailField)
```

Reply:
But I don't see any open tickets / issues for this test failing / being flaky in GitHub.

Reply:
I checked the test history. The test failed twice recently, Aug 20 and Aug 03. 
Do you mind adding the sleep? Thanks for working on it.

Reply:
> Right. We could easily enough not define the resources when the service account isn't created... but there's an explicit `depends` later that I think would be harder to get rid of....
> 

I think we can split the two cases (empty service account and non-empty service account) into its own functions . It is also OK to keep it like this now.

Reply:
pushed that up.

Reply:
> I think we can split the two cases (empty service account and non-empty service account) into its own functions . It is also OK to keep it like this now.

Yeah, I'm going to leave it for now, but looks like it should be pretty easy to adjust the conditional (just might be hard to read), or, as you say, split it into two separate functions.

I'm guessing it won't affect execution time or success rate much, though?

Reply:
> I'm guessing it won't affect execution time or success rate much, though?

I don't think it will affect those.